packages/block-library/src/site-title/edit.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	useEntityProp,
+	useEntitySaving,
+	EntityProvider,
+} from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { RichText } from '@wordpress/block-editor';
+
+function TitleInput() {
+	const [ title, setTitle ] = useEntityProp( 'root', 'site', 'title' );
+	const [ isDirty, isSaving, save ] = useEntitySaving( 'root', 'site', 'title' );
+	return (
+		<>
+			<Button
+				isPrimary
+				className="wp-block-site-title__save-button"
+				disabled={ ! isDirty || ! title }
+				isBusy={ isSaving }
+				onClick={ save }
+			>
+				{ __( 'Update' ) }
+			</Button>
+			<RichText
+				tagName="h1"
+				placeholder={ __( 'Site Title' ) }
+				value={ title }
+				onChange={ setTitle }
+			/>
+		</>
+	);
+}
+
+export default function SiteTitleEdit() {
+	return (
+		<EntityProvider kind="root" type="site">
```

Comment:
Any thoughts about multi-site? I guess it's not possible to have a multi-site post but what about a potential `SiteList` block that renders multiple Sites, and each site having a title and a description.

I guess this is me saying that, don't  you think we need to separate the "Site block" from  the  "Site title block" (same for post and post title, post content).

Reply:
The same question applies to the backend. How would this block work inside a multi-site "SiteList" block.

Reply:
> Under what site and use case would a SiteList render.

Any site from the same network. Granted that there's no site endpoint at the moment but I think it will come.

It's the same discussion for post related blocks. You're in the context of a single post but you can still show a list of posts in a "recent posts block" and we should be able to reuse the same PostTitle, PistContent blocks there.

Reply:
`PostTitle` and `PostContent` are different, they default to the root post, but can also be nested in a `Post` block that provides a different post through an `EntityProvider`.

We can implement the same thing for the `SiteTitle` block when we get a site endpoint.

Reply:
Basically, all the code in these blocks just looks up for the nearest `EntityProvider` so nothing stops us from having wrapper blocks later on.

Reply:
I agree with you that we might want a fallback.

That said, I don't think the block itself should define the `EntityProvider` but it should be something added at the root level of the  BlockEditor (like the default Post EntityProvider)

Reply:
Sure, that makes sense, we could do it later as well and it wouldn't be a breaking change, but it's probably better to just do it now.

Reply:
Much cleaner: https://github.com/WordPress/gutenberg/pull/17207/commits/22ed73ff25238adf7768d21a783f5b55112bac41

Thanks for the suggestion! :smile: